### DISCUSSION TITLE
Join table query returns null if foreign key is added

### DISCUSSION THREAD

**Discussion by Zizazorro** [supabase/supabase#28353] [Questions] [answered]
So I'm learning Supabase, but I got stuck on a query. I reproduced this docs example: https://supabase.com/docs/reference/javascript/select, with the example "Query referenced tables through a join table". I added the example data source in the SQL Editor and created those tables.

I tested it with the example query and it gave the expected reponse:

```
const { data, error } = await supabase
  .from('users')
  .select(`
    name,
    teams (
      name
    )
  `)
```

However, when I added a new column (trainer_id, which is a foreign key to the users.id) in the teams table, that same query returns null.

Why does it return null if a foreign key is added? Should I change my query? The goal is to get the exact same response as in the example.

**Comment by GaryAustin1** [comment, marked as answer]
You may need to look at your error response.  Without you showing what you are adding, it is sort of unclear.  My guess is you have 2 foreign keys between two tables and the API is confused as which one to use.  Usually the error message will give a hint at a fix.

**Reply by Zizazorro** [reply]
You were correct! It was because there were two relationships between users and teams. Thanks a lot!

Answer:

```
const { data, error } = await supabase.from('users').select(`
      name,
      teams!users_teams (
        name
      )
    `);
```
